jlm/llvm/opt/alias-analyses/RegionAwareMemoryNodeProvider.hpp

```diff
@@ -210,7 +191,31 @@ class RegionAwareMemoryNodeProvider final : public MemoryNodeProvider
   static std::string
   ToRegionTree(const rvsdg::Graph & rvsdg, const RegionAwareMemoryNodeProvisioning & provisioning);
 
+  /**
+   * The provisioning produced by this provider
+   */
   std::unique_ptr<RegionAwareMemoryNodeProvisioning> Provisioning_;
+
+  /**
+   * The SCCs of functions, in reverse topological order
+   * If function a() calls b(), and they are not in the same SCC, a()'s SCC comes after b()'s SCC.
+   */
+  std::vector<util::HashSet<const rvsdg::LambdaNode *>> FunctionSCCs_;
+
+  /**
+   * Which SCC contains the node representing external functions
+   */
+  size_t ExternalNodeSccIndex_;
+
+  /**
+   * A mapping from functions to the index of the SCC they belong to in the call graph
+   */
+  std::unordered_map<const rvsdg::LambdaNode *, size_t> FunctionToSccIndex_;
+
+  /**
+   * The set of memory nodes that may be affected within a given SCC. Indexed by sccIndex.
+   */
+  std::vector<util::HashSet<const PointsToGraph::MemoryNode *>> SccSummaries_;
```

Comment:
One thing that just popped up in my mind: The Provision method is expected to be called multiple times and arrive at the same result. However, this would not happen right now as the private attributes are not set up right now. That is also the reason why I always put everything into Context and then just reallocate this.

Reply:
That is a very good point, I can create a Context wrapper